tldr/cfg_extractor.py

```
@@ -637,20 +702,62 @@ class TreeSitterCFGBuilder:
     """
 
     # Node type mappings per language
-    IF_TYPES = {"if_statement", "if_expression", "if"}  # Ruby uses "if"
-    WHILE_TYPES = {"while_statement", "while_expression", "while"}  # Ruby uses "while"
-    FOR_TYPES = {"for_statement", "for_expression", "for_in_statement", "for", "foreach_statement", "for_range_loop", "for_generic_clause", "for_numeric_clause"}  # Ruby uses "for", PHP uses "foreach_statement", C++ uses "for_range_loop", Lua uses for_generic/for_numeric
+    IF_TYPES = {"if_statement", "if_expression", "if"}  # Ruby uses "if", R uses "if"
```

Ruff (0.14.13)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)